### ISSUE TITLE
Class name handling for Collections.emptyMap() changed in 2.9.3+

### ISSUE THREAD

**Issue by rwinch** [FasterXML/jackson-databind#1881]
Similar to #1868  the following test works in jackson-databind 2.9.2 and prior, but breaks in 2.9.3 and 2.9.4-SNAPSHOT

```java
@Test
public void serializeEmptyMap() throws Exception {
	ObjectMapper mapper = new ObjectMapper();
	mapper.enableDefaultTyping(ObjectMapper.DefaultTyping.NON_FINAL, JsonTypeInfo.As.PROPERTY);
	String actualJson = mapper.writeValueAsString(Collections.emptyMap());
	assertEquals("{\"@class\":\"java.util.Collections$EmptyMap\"}", actualJson);
}
```

with the following

```
org.junit.ComparisonFailure: 
Expected :{"@class":"java.util.Collections$EmptyMap"}
Actual   :{"@class":"java.util.HashMap"}
```

**Comment by cowtowncoder** [comment]
Hmmh. Now this is also dangerous one. But, I do think that preserving existing behavior (and not trying for "unrequested fix" here), that is, 2.9.2, not modifying class name, is the least bad option.

With 3.x probably makes sense to somehow connect this to other "change the class name on the fly" functionality (white-listing). Or, if deemed worth it, custom deserializer for these types. Even if JDK does not offer real help in determining such add-on qualifiers (empty, singleton, unmodifiable).

**Comment by rwinch** [comment]
I should add that this is also happening in 2.8.11.1-SNAPSHOT

**Comment by cowtowncoder** [comment]
Fixed along with #1880 -- see notes on that one.
